src/app/apps/lti/views.py

```diff
@@ -1,26 +1,60 @@
 """Views for the LTI app."""
 
+import json
 import logging
+from urllib.parse import unquote
 
 from django.conf import settings
 from django.core.exceptions import PermissionDenied
 from django.http import HttpRequest, HttpResponse
-from django.shortcuts import redirect
-from django.views.generic import TemplateView
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+from django.views.generic.base import TemplateResponseMixin, TemplateView
 from lti_toolbox.exceptions import LTIException
+from lti_toolbox.launch_params import LTIMessageType
 from lti_toolbox.lti import LTI
+from lti_toolbox.models import LTIPassport
 from lti_toolbox.views import BaseLTIView
+from oauthlib import oauth1
 
 from .forms import BaseLTIUserForm
 
 logger = logging.getLogger(__name__)
 
 
-class LTIRequestView(BaseLTIView):
+class RenderMixins(TemplateResponseMixin, View):
+    """Mixin class for rendering an LTI view with app_data.
+
+    This mixin provides the necessary template to render the frontend
+    while passing app_data through the DOM. By calling the 'render' method,
+    the app_data is constructed. The default value for app_data is an
+    empty dictionary, which can be overridden in your subclass.
+
+    Attributes:
+        template_name (str): The name of the template to be used for rendering.
+        app_data (dict): Data to be passed to the frontend. Default is an empty dict.
+    """
+
+    template_name = "base.html"
+    app_data = {}
+
+    def render(self):
+        """Render view's template with app's data.
+
+        Returns:
+            HttpResponse: The HTTP response containing the rendered template.
+        """
+        context = {"app_data": json.dumps(self.app_data)}
+        return self.render_to_response(context)
```

Comment:
I thinks it's more pythonic to only override the `TemplateResponseMixin` (you don't need to inherit from the View class):

```suggestion
class RenderMixins(TemplateResponseMixin):
    """Mixin class for rendering an LTI view with app_data.

    This mixin provides the necessary template to render the frontend
    while passing app_data through the DOM. By calling the 'render' method,
    the app_data is constructed. The default value for app_data is an
    empty dictionary, which can be overridden in your subclass.

    Attributes:
        template_name (str): The name of the template to be used for rendering.
        app_data (dict): Data to be passed to the frontend. Default is an empty dict.
    """

    template_name = "base.html"
    app_data = {}

    def render_to_response(self, context, **response_kwargs):
        """Render view's template with app's data.

        Returns:
            HttpResponse: The HTTP response containing the rendered template.
        """
        context.update({"app_data": json.dumps(self.app_data)})
        return super().render_to_response(context, **response_kwargs)
```

Reply:
That's a very good point !! thx ☺️